core/js/src/object_util.ts

```diff
@@ -164,3 +164,21 @@ export function getRecordKeys<T extends Record<string | number | symbol, any>>(
 ): Array<keyof T> {
   return Object.keys(obj);
 }
+
+/**
+ * Get object value by providing a path to it as an array.
+ * @example getObjValueByPath({ bar: { foo: "Hello, world!" } }, ["bar", "foo"]); // "Hello, world!"
+ */
+export function getObjValueByPath(
+  row: Record<string, unknown>,
+  path: string[],
+): unknown {
+  let curr: unknown = row;
+  for (const p of path) {
+    if (!isObjectOrArray(curr)) {
+      return null;
```

Comment:
nit: probably want to return `undefined` to disambiguate from a path with a value of `null`